include/smear.h

```diff
@@ -24,6 +27,40 @@
         SRT_send_message(msg, CAT(m, _handler));                       \
     }
 
+/* cancel_token_t SRT_delayed_send(<machine name>, <event name>, void *msg,
+                                   uint64_t delay_ms); */
```

Comment:
Yeah, it's pretty magical. I really didn't want to make people build their own wrappers though, and couldn't see a way to achieve that without something like this.

Code gen could accomplish it, but it would take some doing. We'd have to generate `<machine>_<event>_after(<event>_t *e, delay_ms)` and get pretty tightly coupled with the implementation.

Reply:
Yeah, it could be that, or we could have some kinda backend magic, maybe a flag bit in the wrapper denoting that the event was cancellable, or something?  Then the regular functions might be usable somehow.  Incomplete thought, that.

Reply:
The issue is that if an event is cancellable the function to send it must return something. That something is tightly coupled to the implementation.

Reply:
This will wind up being consumed by generated code as well, in the form of a `cancellable` machine.  What would that code look like?  That might influence whether this is a good design or not.

I think one of my hangups is that it forces the call site to have names as arguments, which makes it impossible to be generic.

Reply:
Now I have to think about things. Maybe the right way to do this is with code gen after all.